install.sh

```diff
@@ -180,47 +167,62 @@ if [[ ! -d "$INSTALL_PATH" ]] || [[ ! -w "$INSTALL_PATH" ]]; then
     else
       abort "Could not find or create a writable installation directory. Try running with sudo or specify a different directory with --dir."
     fi
-  else
-    abort "No write permission to $INSTALL_PATH. Try running with sudo or specify a different directory with --dir."
   fi
-fi
+}
 
-TMP_DIR="$(mktemp -d)"
-cleanup() {
-  rm -rf "$TMP_DIR"
+check_version() {
+    if [[ -z "$VERSION" ]]; then
+      abort "Version is empty. This should not happen."
+    fi
 }
-trap cleanup EXIT
-
-if [[ "$VERSION" == "latest" ]]; then
-  ohai "Fetching latest release information..."
-  RELEASE_URL="https://agentuity.sh/release/cli"
-  VERSION=$(curl -s $RELEASE_URL | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
-  if [[ -z "$VERSION" ]]; then
-    abort "Failed to fetch latest version information"
+
+check_latest_release() {
+  if [[ "$VERSION" == "latest" ]]; then
+    ohai "Fetching latest release information..."
+    RELEASE_URL="https://agentuity.sh/release/cli"
+    VERSION=$(curl -s $RELEASE_URL | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+    if [[ -z "$VERSION" ]]; then
+      abort "Failed to fetch latest version information"
+    fi
   fi
-fi
+}
 
-VERSION="${VERSION#v}"
+install_mac() {
+  if command -v brew >/dev/null 2>&1 && [[ "$NO_BREW" != "true" ]]; then
+    ohai "Homebrew detected! Installing Agentuity CLI using Homebrew..."
+      
+    if [[ "$VERSION" != "latest" ]]; then
+      ohai "Installing Agentuity CLI version $VERSION using Homebrew..."
+      brew install agentuity/tap/agentuity@${VERSION#v}
+    else
+      ohai "Installing latest Agentuity CLI version using Homebrew..."
+      brew install -q agentuity/tap/agentuity
+    fi
 
-if [[ "$OS" == "Windows" ]]; then
-  if [[ "$ARCH" == "x86_64" ]]; then
-    DOWNLOAD_FILENAME="agentuity-x64.msi"
-  elif [[ "$ARCH" == "arm64" ]]; then
-    DOWNLOAD_FILENAME="agentuity-arm64.msi"
-  else
-    DOWNLOAD_FILENAME="agentuity-x86.msi"
+    if command -v agentuity >/dev/null 2>&1; then
+      success
+    else
+      abort "Homebrew installation failed. Please try again or use manual installation."
+    fi
+
+    else
+      ohai "Installing Agentuity CLI using curl..."
+      curl -fsSL https://agentuity.sh/install.sh | bash
   fi
-else
-  DOWNLOAD_FILENAME="agentuity_${OS}_${ARCH}.${EXTENSION}"
-fi
 
-DOWNLOAD_URL="https://github.com/agentuity/cli/releases/download/v${VERSION}/${DOWNLOAD_FILENAME}"
-CHECKSUMS_URL="https://github.com/agentuity/cli/releases/download/v${VERSION}/checksums.txt"
+}
+
+download_release() {
 
-ohai "Downloading Agentuity CLI v${VERSION} for ${OS}/${ARCH}..."
-curl -L --progress-bar "$DOWNLOAD_URL" -o "$TMP_DIR/$DOWNLOAD_FILENAME" || abort "Failed to download from $DOWNLOAD_URL"
+  DOWNLOAD_FILENAME="agentuity_${OS}_${ARCH}.${EXTENSION}"
+  DOWNLOAD_URL="https://github.com/agentuity/cli/releases/download/v${VERSION#v}/${DOWNLOAD_FILENAME}"
```

Comment:
this needs to use agentuity.sh

Reply:
PR pending: https://github.com/agentuity/get-agentuity-sh/pull/3